news/2020-04-01-How-to-configure-wildfly-s2i-by-cli-operations.adoc

```diff
@@ -0,0 +1,446 @@
+= Configuring WildFly S2I image by using CLI Management Operations
+yborgess
+2020-04-01
+:awestruct-tags: [wildfly, galleon]
+:awestruct-layout: blog
+:source-highlighter: coderay
+:encoding: utf-8
+:lang: en
+
+== Introduction
+
+The standard and recommended way to configure the WildFly cloud images is by using environment variables. However, you could find useful for your use case to configure the server by using a custom CLI https://docs.wildfly.org/19/Admin_Guide.html#operations[management operations] script.
+
+The following post describes how you can apply management operations to configure the WildFly server image. We will show you how you can execute CLI scripts at S2I phase and how to use the extensions mechanism provided by the WildFly cloud image to execute CLI management operations at runtime.
+
+This practical guide uses https://github.com/code-ready/crc[Red Hat CodeReady Containers] as a local Openshift cluster. It assumes you have basic knowledge on Openshift and you know how to configure WildFly S2I image by using environment variables.
+
+== Executing a CLI management operations script at Source-to-Image (S2I) phase
+
+https://github.com/openshift/source-to-image[Source-to-Image (S2I)] is the tool used internally by Openshift to build container images from application source code. When we are creating an Openshift new application using the WildFly image stream, S2I takes our application source code from a Git repository, provisions the WildFly server by using Galleon layers and builds the final image that runs the assembled application.
+
+In this example we are going to assemble this https://github.com/yersan/jaxrs-postgresql-demo[JAX-RS PostgreSQL] demo application with the WildFly server provisioned by Galleon with the PostgreSQL drivers. Our demo application expects a data source available under the following `java:jboss/datasources/PostgreSQLDS` JNDI resource. This data source will be configured executing the following https://github.com/yersan/jaxrs-postgresql-demo/blob/cli-at-s2i/s2i-config/config-ds.cli[config-ds.cli] CLI script:
```

Comment:
This example is very nice, would it be possible to have this example to be located in https://github.com/wildfly/wildfly-s2i/tree/master/examples ? With a README.md with the minimal set of steps needed to run it in openshift?

Reply:
If it gets approved, I can create a PR to include it as an example in wildfly-s2i.